homeassistant/components/subaru/entity.py

```diff
@@ -0,0 +1,40 @@
+"""Base class for all Subaru Entities."""
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .const import DOMAIN, ICONS, VEHICLE_NAME, VEHICLE_VIN
+
+
+class SubaruEntity(CoordinatorEntity):
+    """Representation of a Subaru Entity."""
+
+    def __init__(self, vehicle_info, coordinator):
+        """Initialize the Subaru Entity."""
+        super().__init__(coordinator)
+        self.car_name = vehicle_info[VEHICLE_NAME]
+        self.vin = vehicle_info[VEHICLE_VIN]
+        self.title = "entity"
+
+    @property
+    def name(self):
+        """Return name."""
+        return f"{self.car_name} {self.title}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return slugify(f"{DOMAIN} {self.vin} {self.title}")
```

Comment:
```suggestion
        return f"{self.vin}_{self.title}"
```

Is title stable?

Reply:
Yes, title is stable.

Reply:
Please rename `self.title` to something that implies it is not configurable by the user.  Right now `title` implies its a name, which will make this more difficult to review in the future as it will trigger the reviewer to check to make sure its a stable identifier.

Maybe `self.XXXX_type`

Reply:
changed to `self.entity_type`